File: app/src/main/java/com/snatik/matches/themes/Themes.java
package com.snatik.matches.themes;

import android.graphics.Bitmap;

import com.snatik.matches.common.Shared;
import com.snatik.matches.utils.Utils;

import java.util.ArrayList;

public class Themes {

    public static String URI_DRAWABLE = "drawable://";

    public static Theme createTheme(@Theme.ThemeId int themeId) {
        switch (themeId) {
            case Theme.ID_ANIMAL:
            case Theme.ID_ANIMAL_VISUAL:
            case Theme.ID_ANIMAL_AUDITORY:
                Theme animalTheme = createAnimalsTheme();
                animalTheme.id = themeId;
                return animalTheme;
            case Theme.ID_MONSTER:
                return createMosterTheme();
            case Theme.ID_EMOJI:
                return createEmojiTheme();
            default:
                throw new IllegalArgumentException("Theme " + themeId + " not found.");
        }
    }


    public static Theme createAnimalsTheme() {
        Theme theme = new Theme();
        theme.id = Theme.ID_ANIMAL;
        theme.name = "Animals";
        theme.backgroundImageUrl = URI_DRAWABLE + "back_animals";
        theme.tileImageUrls = new ArrayList<>();
        // 40 drawables
        for (int i = 1; i <= 28; i++) {
            theme.tileImageUrls.add(URI_DRAWABLE + String.format("animals_%d", i));
        }
        return theme;
    }

    public static Theme createMosterTheme() {
        Theme theme = new Theme();
        theme.id = Theme.ID_MONSTER;
        theme.name = "Mosters";
        theme.backgroundImageUrl = URI_DRAWABLE + "back_horror";
        theme.tileImageUrls = new ArrayList<>();
        // 40 drawables
        for (int i = 1; i <= 40; i++) {
            theme.tileImageUrls.add(URI_DRAWABLE + String.format("mosters_%d", i));
        }
        return theme;
    }

    public static Theme createEmojiTheme() {
        Theme theme = new Theme();
        theme.id = Theme.ID_EMOJI;
        theme.name = "Emoji";
        theme.backgroundImageUrl = URI_DRAWABLE + "background";
        theme.tileImageUrls = new ArrayList<>();
        // 40 drawables
        for (int i = 1; i <= 40; i++) {
            theme.tileImageUrls.add(URI_DRAWABLE + String.format("emoji_%d", i));
        }
        return theme;
    }

    public static Bitmap getBackgroundImage(Theme theme) {
        String drawableResourceName = theme.backgroundImageUrl.substring(Themes.URI_DRAWABLE.length());
        int drawableResourceId = Shared.context.getResources().getIdentifier(drawableResourceName, "drawable", Shared.context.getPackageName());
        Bitmap bitmap = Utils.scaleDown(drawableResourceId, Utils.screenWidth(), Utils.screenHeight());
        return bitmap;
    }

}

File: app/src/main/java/de/rafaelmiranda/memory/themes/Themes.kt
package de.rafaelmiranda.memory.themes

import android.graphics.Bitmap
import de.rafaelmiranda.memory.common.Shared
import de.rafaelmiranda.memory.utils.Utils
import java.util.*


object Themes {

    const val URI_DRAWABLE = "drawable://"

    private const val NORMAL_SET_URI = "animals_%d"
    private const val BLUR_SET_URI = "animals_blur_%d"

    fun createTheme(@Theme.ThemeId themeId: Int): Theme {
        val theme = Theme()
        theme.id = themeId
        theme.name = "Animals"
        theme.tileImageUrls = createCardsSet(themeId)
        return theme
    }

    private fun createCardsSet(themeId: Int): List<String> {
        val tileImageUrls = ArrayList<String>()
        val baseCardsUri = if (themeId == Theme.ID_VISUAL_BLUR) BLUR_SET_URI else NORMAL_SET_URI
        // 40 drawables
        for (i in 1..28) {
            tileImageUrls.add(URI_DRAWABLE + String.format(baseCardsUri, i))
        }
        return tileImageUrls
    }


    fun getBackgroundImage(theme: Theme): Bitmap {
        val drawableResourceName = theme.backgroundImageUrl.substring(Themes.URI_DRAWABLE.length)
        val drawableResourceId = Shared.context.resources.getIdentifier(drawableResourceName, "drawable", Shared.context.packageName)
        return Utils.scaleDown(drawableResourceId, Utils.screenWidth(), Utils.screenHeight())
    }

}
